src/metabase/task/notification.clj

```diff
@@ -104,16 +109,50 @@
+  []
+  (let [triggers     (-> send-notification-job-key task/job-info :triggers)
+        new-timezone (send-notification-timezone)
+        subscription-id->cron (t2/select-fn->fn :id :cron_schedule :model/NotificationSubscription :type :notification-subscription/cron)]
+    (doseq [trigger triggers
+            :when (not= new-timezone (:timezone trigger))] ; skip if timezone is the same
+      (let [trigger-key     (:key trigger)
+            subscription-id (send-notification-trigger-key->subscription-id trigger-key)]
+        (log/infof "Updating timezone of trigger %s to %s. Was: %s" trigger-key new-timezone (:timezone trigger))
+        (task/reschedule-trigger! (build-trigger subscription-id (get subscription-id->cron subscription-id)))))))
+
 (jobs/defjob ^{:doc "Triggers that send a notification for a subscription."}
   SendNotification
   [context]
   (let [{:strs [subscription-id]} (qc/from-job-data context)]
     (send-notification* subscription-id)))
 
+(jobs/defjob
+  ^{:doc
+    "Find all notification subscriptions with cron schedules and create a trigger for each.
+
+    Context: We've migrated alerts from pulse to notifications, see the `v53.2024-12-12T08:05:00` migration.
+    This job is needed to create triggers for all existing notification subscriptions.
+    Will only run once when Metabase starts up."}
+  InitNotificationTriggers
+  [_context]
+  (run! update-subscription-trigger! (t2/reducible-select :model/NotificationSubscription)))
+
 (defmethod task/init! ::SendNotifications [_]
-  (let [send-notification-job (jobs/build
-                               (jobs/with-identity send-notification-job-key)
-                               (jobs/with-description "Send Notification")
-                               (jobs/of-type SendNotification)
-                               (jobs/store-durably))]
-    (task/add-job! send-notification-job)))
+  (let [send-notification-job              (jobs/build
+                                            (jobs/with-identity send-notification-job-key)
+                                            (jobs/with-description "Send Notification")
+                                            (jobs/of-type SendNotification)
+                                            (jobs/store-durably))
+        init-notification-triggers-job     (jobs/build
+                                            (jobs/of-type InitNotificationTriggers)
+                                            (jobs/with-identity (jobs/key "metabase.task.notification.init-notification-triggers.job"))
+                                            (jobs/store-durably))
+        init-notification-triggers-trigger (triggers/build
+                                            (triggers/with-identity (triggers/key "metabase.task.notification.init-notification-triggers.trigger"))
+                                            ;; run only once per MB instance (like a migration)
+                                            (triggers/start-now))]
```

Comment:
I just want to double check this doesn't cause us to rerun when the process restarts. My mental models says that when we run `schedule-task!` it is going to delete the old trigger, insert this new instance, and run again.

I wouldn't be shocked if my mental model is wrong though, still a Quartz noob 😄